src/components/workspace/qr-gen.tsx

```diff
@@ -0,0 +1,30 @@
+import { createEffect, createSignal } from "solid-js";
+import * as qrcode from "qrcode";
+
+export default function CertQrCode({ value }: { value: string }) {
+  const [qrCodeData, setQRCodeData] = createSignal<string | null>(null);
```

Comment:
`|null` seems to be uncommon. I'm not familiar but I think one of the following is better:
```typescript
const [qrCodeData, setQRCodeData] = createSignal('')  // Always <string> type
```
```typescript
const [qrCodeData, setQRCodeData] = createSignal<string>()  // <undefined> by default
```